### ISSUE TITLE
draft copyediting file copied from a discussion can not be accessed

### ISSUE THREAD

**Issue by bozana** [pkp/pkp-lib#10584]
S. https://forum.pkp.sfu.ca/t/assoc-type-preventing-file-download-by-copyeditor-ojs-3-3-0-13/93501

When using the function `Upload/Select File` for `Draft Files` in `Copyediting` tab the users can select files from `Copyediting Discussions`. 
1) Also the copyeditor that is not a participant of (and thus has no access to) the discussion, that the discussion file belongs to, can select the discussion file and add it to the `Draft Files` area.
2) The discussion file is copied to the Draft Files area, but it keeps the assoc_type = 520 (which means discussion file).
It results that the copyeditor that has no access to the discussion and discussion file see the new file in the `Draft Files` area, but he/she cannot download the file.

I believe here are two things to consider:
- The users that have no access to discussion files should not be able to select them from the menu `Upload/Select File`.
- I would think that a file that is selected and displayed in the `Draft Files` area should not get the assoc_type = 520 i.e. that is should be accessed by everybody with the access to that area.

Tested on OJS 3.3. Needs to be tested for 3.4 and main.

**Comment by bozana** [comment]
Yes, that could also be, but also a copyeditor not involved in the discussion - that does not have access to the discussion and file - could select the file and copy to the draft section...

**Comment by asmecher** [comment]
@bozana, were you able to reproduce this one?
